src/app/home/register-winner/register1vs1-match/register1vs1-match.component.ts

```diff
@@ -39,34 +39,34 @@ export class Register1vs1MatchComponent implements OnInit {
   }
 
   registerWinner() {
-    if (this.player1 && this.player2) {
-      if (confirm('Did ' + this.player1.name + ' win?')) {
-        this.isLoading = true;
-        const matchModel: MatchModel = this.createMatch(this.player1, this.player2);
-        this.matchmaker.registerWinner(true, matchModel).then(result => {
-          this.isLoading = false;
-          if (result) {
-            this.matchResult = result;
-            this.hasRegisterd = true;
-          }
-        });
-      } else {
-        console.log('Cancel');
-      }
+    if (!(this.player1 && this.player2)) {
+      return;
+    }
+    if (!confirm('Did ' + this.player1.name + ' win?')) {
```

Comment:
I would prefer if a new variable was introduced here. Also, template strings are a bit easier to read than string concatenation. I'd do something like 

```javascript
const player1Won = confirm(`Did ${this.player1.name} win?`);
if (!player1Won) {
  // ...
}
```

however, this is your call